### PR TITLE
Properly handle the Python version when looking for Python binary

### PR DESCRIPTION
**Pull request by 0danteh**
The second code snippet is a simplified version of the first one. It does not use a custom function, but instead uses the built-in `os.environ.get` function to get the value of the `PYTHON_BIN_PATH` environment variable, or the default value if it is not set. It also uses the built-in `shutil.which` function to find the path of the Python binary in the system path. This function returns `None` if the path is not found, so the code checks for that and raises a `FileNotFoundError` exception. It also checks the Python version by using another custom function called `get_python_major_version` and raises a `ValueError` exception if the version is not 3. If any exception is raised, the code exits with an error code. Otherwise, it sets the `PYTHON_BIN_PATH` environment variable to the valid path and converts it to Windows style if needed.

The main differences between the original snippet and the edited one are:

- The original code snippet uses a custom function to get the Python binary path, while the second code snippet uses built-in functions.
- The original code snippet asks the user to input the Python binary path if it is not found or invalid, while the second code snippet raises an exception and exits.
- The original code snippet does not check the Python version, while the second code snippet does and requires it to be 3.
- The original code snippet ignores the `PYTHON_BIN_PATH` environment variable if it is already set, while the second code snippet respects it.

Why I have considered committing these changes: 

- It is more concise and readable, using fewer lines of code and avoiding unnecessary custom functions.
- It is more robust and consistent, using standard functions that handle edge cases and errors, and respecting the existing environment variables.
- It is more secure and compatible, checking the Python version and ensuring it is 3, which is the recommended version for most Python projects.

### CODE REVIEW

**Review by mihaimaruseac — COMMENTED** [review]
This seems reasonable, but can you please try updating PR title, PR message (both of this via direct edits from the interface) and commit message to be more descriptive, please? This way the PR will have a faster turn-around time on internal systems, and people looking at history can understand at a glance what this does.

To edit change the PR title and the PR message, just click the corresponding edit buttons on the interface.

To change the commit, try `git commit --amend` on the branch you have the commit in and then `git push -f` (assuming using CLI -- if you use a visual tool around git, please consult the tool's documentation).

Thank you!

**Review by mihaimaruseac — APPROVED** [review]
Thank you! This looks good!

There is a pylint failure on 3 lines being too long. It might be worthwhile to trim them to 80 characters.

**Review by mihaimaruseac — APPROVED** [review]

**Review by mihaimaruseac — COMMENTED** [review]
There are 2 strays parentheses, fixing via suggested edits

**Review by mihaimaruseac — APPROVED** [review]

**Review by mihaimaruseac — COMMENTED** [review]
Using f-strings to shorten the lines

**Review by mihaimaruseac — APPROVED** [review]

File: configure.py
Comment (mihaimaruseac) [inline configure.py:198]:
```suggestion
        raise FileNotFoundError(f'Python path {python_bin_path} is invalid.')
```

File: configure.py
Comment (mihaimaruseac) [inline configure.py:201]:
```suggestion
      raise ValueError(f'Python from {python_bin_path} is not version 3.')
```

**Comment by mihaimaruseac** [comment]
I added a few changes via suggested edits to make this pass internal tests and land.

Thank you very much for the PR!

**Review by mihaimaruseac — APPROVED** [review]

**Comment by mihaimaruseac** [comment]
I think for now it should be good. There are a few CI jobs that take longer and then there needs to be another approver in the internal system (2 pair of eyes required on any change to prevent insider risk). After that, this will merge automatically.

Thank you!

**Comment by angerson** [comment]
Hi, thanks for your work as a first-time OSS contributor. I can see that you're trying to make the code more resilient and readable -- but these changes are not useful, e.g.:

- They make this specific function behave much differently than everything else in this configure.py script, which is confusing for anyone trying to understand the whole script
- There is no reason to remove the utility function
- Removing the loop means that users have only one chance to get the path right, or else the script quits (this behavior should instead be changed explicitly and across the whole script)

I don't see what specific problem this change will fix, but instead, it makes the code, overall, less consistent. I'd like you to reduce the change to **only fix a bug that you found and nothing else.** I see that you originally indicated that this change would fix a bug where PYTHON_BIN_PATH would be ignored. What do you mean by that?

**Comment by dante-tech** [comment]
I see your perspective, let me clarify your points further.

- **First**: the commit made is done purposefully using standard functions that handle edge cases and errors, also, by respecting the existing environment variables.

- **Second**: built-in functions are more efficient and optimized than custom functions. They are implemented in C or other low-level languages and can run faster and use less memory than custom functions. Custom functions may have redundant code that slows down the execution or wastes resources. Therefore, even for just one instance, I preferred to use built-in functions over the custom one.

- **Third**: it manages to solve a bug. The bug I'm trying to solve and addressing, is related to the `PYTHON_BIN_PATH` environment variable. The original code snippet ignores the `PYTHON_BIN_PATH` environment variable if it is already set. This means that it does not respect the user’s preference and may use a different Python binary than the one the user intended. This can cause problems if the user wants to use a specific version of Python or a virtual environment. This is also raised by one of the latest issues. My changes respects the `PYTHON_BIN_PATH` environment variable and uses it as the default value. This means that it will use the Python binary that the user specified, unless it is not valid or not found. This way, the user can control which Python binary to use and avoid potential conflicts or errors.

I hope this clarifies it all, have a good day!

**Comment by mihaimaruseac** [comment]
> * built-in functions are more efficient and optimized than custom functions. They are implemented in C or other low-level languages and can run faster and use less memory than custom functions

This seems weird. There is no C involved here, if I understand correctly

**Comment by dante-tech** [comment]
@mihaimaruseac I was referring myself to the built-in functions, not the code

**Comment by angerson** [comment]
Thanks for clarifying. This change is not necessary.

**Comment by dante-tech** [comment]
Thank you for your time, however there's something I'm missing @angerson; why the change is not necessary?

**Pull request closed without merging**
